DelvUI/Interface/HudHelper.cs

```diff
@@ -271,9 +275,15 @@ private void UpdateCombatActionBars(bool forceUpdate = false)
             {
                 currentCombatState = !currentCombatState && Config.ShowCombatActionBarsOnWeaponDrawn
                     ? HasWeaponDrawn() || IsInCombat()
-                    : Config.ShowCombatActionBarsInDuty
-                        ? IsInDuty() || IsInCombat()
-                        : IsInCombat();
+                    : currentCombatState;
+                
+                currentCombatState = !currentCombatState && Config.ShowCombatActionBarsWhenCrafting
+                    ? IsCrafting() || IsInCombat()
+                    : currentCombatState;
+                
+                currentCombatState = !currentCombatState && Config.ShowCombatActionBarsWhenGathering
+                    ? IsGathering() || IsInCombat()
+                    : currentCombatState;
```

Comment:
This doesn't seem right.
ShowCombatActionBarsInDuty and IsInDuty are not being used anymore unless I missed something.